app/controllers/pairings_controller.rb

```diff
@@ -0,0 +1,78 @@
+class PairingsController < ApplicationController
+  before_action :require_current_user
+
+  def new
+    #find any existing pending_pairings
+    @user = User.find_by(id: params[:user_id])
+    potential_pairing = @user.find_random_pending_pair_from_friends
+    potential_pairing ? @pairing = potential_pairing : @pairing = generate_pair
+    unless @pairing
+      flash[:warn] = "Unable to generate a new pair. Try adding some friends!"
+      redirect_to :back
+    end
+  end
+
+  def create
+    @user = User.find_by(id: params[:user_id])
+    @pairing = Pairing.new(pairing_params)
+    if @pairing.save
+      @match = Match.create(user_id: current_user.id)
```

Comment:
This feels like core-ish logic that maybe should be in a model (like an after_save on pairing.)